src/cli.c

```diff
@@ -134,7 +139,11 @@ int main(int argc, char **argv) {
         } else {
             const char *filename;
             filename = argv[optind];
-            JSValue ret = TJS_EvalFile(ctx, filename, JS_EVAL_TYPE_MODULE, true);
+            int flags = JS_EVAL_TYPE_MODULE;
+            if (strict_module_detection && has_suffix(filename, ".mjs")) {
```

Comment:
Isn't this the opposite of what you want here? you probably want `!has_suffix`

Reply:
Fixed!